geoapps/edge_detection/application.py

```diff
@@ -191,30 +187,34 @@ def window_size(self):
         return self._window_size
 
     def trigger_click(self, _):
-        entity, _ = self.get_selected_entities()
-        if getattr(self.trigger, "vertices", None) is not None:
-            name = string_name(self.export_as.value)
-            temp_geoh5 = f"{string_name(self.export_as.value)}_{time():.3f}.geoh5"
-            with self.get_output_workspace(
-                self.export_directory.selected_path, temp_geoh5
-            ) as workspace:
-                out_entity = ContainerGroup.create(
-                    workspace,
-                    name=self.ga_group_name.value,
-                    uid=self._unique_object.get(self.ga_group_name.value, None),
-                )
-                curve = Curve.create(
-                    workspace,
-                    name=name,
-                    vertices=self.trigger.vertices,
-                    cells=self.trigger.cells,
-                    parent=out_entity,
-                    uid=self._unique_object.get(name, None),
-                )
-                self._unique_object[name] = curve.uid
-                self._unique_object[self.ga_group_name.value] = out_entity.uid
+
+        param_dict = self.get_param_dict()
+        temp_geoh5 = f"{string_name(self.params.export_as)}_{time():.3f}.geoh5"
+        new_workspace = self.get_output_workspace(
+            self.export_directory.selected_path, temp_geoh5
+        )
+        for key, value in param_dict.items():
+            if isinstance(value, ObjectBase):
+                param_dict[key] = value.copy(parent=new_workspace, copy_children=True)
+
+        param_dict["geoh5"] = new_workspace
+
         if self.live_link.value:
-            monitored_directory_copy(self.export_directory.selected_path, out_entity)
+            param_dict["monitoring_directory"] = self.monitoring_directory
+
+        ifile = InputFile(
+            ui_json=self.params.input_file.ui_json,
+            validation_options={"disabled": True},
+        )
+
+        new_params = EdgeDetectionParams(input_file=ifile, **param_dict)
+        new_params.write_input_file()
```

Comment:
Let's see if we can just use

```
self.params.update(param_dict)
new_params.write_input_file()
```
This would be faster then re-creating an input file from scratch.